### ISSUE TITLE
[4.3[ [4.4] [5.0] (Blog) Sample Data installation broken when "Debug System" is switched on

### ISSUE THREAD

**Issue by richard67** [joomla/joomla-cms#41045]
### Steps to reproduce the issue

Have PHP 8.1 or 8.2.
Make sure to log PHP errors into a log file.

1. Make a new installation with the current 4.3-dev or 4.4-dev or 5.0-dev.
2. Switch on "Debug System" and set "Error Reporting" to "Maximum".
3. Install "Sample Data Blog".

### Expected result

Works.

### Actual result

Fails.

![j5-sample-data-error_2023-06-25](https://github.com/joomla/joomla-cms/assets/7413183/5680d3dc-8c6d-4491-8376-7efa23784ba1)

In the PHP error log file (example from 4.3-dev, but on 4.4-dev and 5.0-dev it's more or less the same, except of different line numbers):
```
Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 51921439 bytes) in /libraries/vendor/joomla/database/src/Mysqli/MysqliStatement.php on line 435, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Stack trace:, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  1. Joomla\\CMS\\Session\\Storage\\JoomlaStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  2. Joomla\\Session\\Storage\\NativeStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:136, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  3. session_write_close() /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  4. Joomla\\Session\\Handler\\DatabaseHandler->write($session_id = 'pf869mmjg4fj6pj9sh1eab3r1m', $session_data = 'joomla|s:51921352:"...') /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  5. Joomla\\Database\\DatabaseDriver->execute() /libraries/vendor/joomla/session/src/Handler/DatabaseHandler.php:318, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  6. Joomla\\Database\\Mysqli\\MysqliStatement->execute($parameters = *uninitialized*) /libraries/vendor/joomla/database/src/DatabaseDriver.php:675, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  7. mysqli_stmt->execute() /libraries/vendor/joomla/database/src/Mysqli/MysqliStatement.php:435, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 51924992 bytes) in Unknown on line 0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Stack trace:, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  1. Joomla\\CMS\\Session\\Storage\\JoomlaStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  2. Joomla\\Session\\Storage\\NativeStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:136, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  3. session_write_close() /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  4. Joomla\\Session\\Handler\\DatabaseHandler->write($session_id = 'pf869mmjg4fj6pj9sh1eab3r1m', $session_data = 'joomla|s:51921352:"...') /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  5. Joomla\\Database\\DatabaseDriver->execute() /libraries/vendor/joomla/session/src/Handler/DatabaseHandler.php:318, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  6. Joomla\\Database\\Mysqli\\MysqliStatement->execute($parameters = *uninitialized*) /libraries/vendor/joomla/database/src/DatabaseDriver.php:675, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  7. mysqli_stmt->execute() /libraries/vendor/joomla/database/src/Mysqli/MysqliStatement.php:435, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Warning:  Unknown: Cannot call session save handler in a recursive manner in Unknown on line 0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Stack trace:, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  1. Joomla\\CMS\\Session\\Storage\\JoomlaStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  2. Joomla\\Session\\Storage\\NativeStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:136, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  3. session_write_close() /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Fatal error:  Allowed memory size of 268435456 bytes exhausted (tried to allocate 207685524 bytes) in Unknown on line 0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
Stack trace:, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  1. Joomla\\CMS\\Session\\Storage\\JoomlaStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:0, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  2. Joomla\\Session\\Storage\\NativeStorage->close() /libraries/src/Session/Storage/JoomlaStorage.php:136, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  3. session_write_close() /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  4. Joomla\\Session\\Handler\\DatabaseHandler->write($session_id = 'pf869mmjg4fj6pj9sh1eab3r1m', $session_data = 'joomla|s:51921352:"...') /libraries/vendor/joomla/session/src/Storage/NativeStorage.php:114, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  5. Joomla\\Database\\DatabaseDriver->execute() /libraries/vendor/joomla/session/src/Handler/DatabaseHandler.php:318, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  6. Joomla\\Database\\Mysqli\\MysqliStatement->execute($parameters = *uninitialized*) /libraries/vendor/joomla/database/src/DatabaseDriver.php:675, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
  7. mysqli_stmt->execute() /libraries/vendor/joomla/database/src/Mysqli/MysqliStatement.php:435, referer: https://www.joomla-43-dev.vmubu01.vmnet2.local/administrator/index.php
```

### System information (as much as possible)

PHP 8.1
Current 4.3-dev or 4.,4-dev or 5.0-dev branch.

**Comment by Fedik** [comment]
Try the fix #39526

**Comment by richard67** [comment]
> Try the fix #39526

@Fedik Ah, that was the one I had in mind, but I haven't found it. Thanks for checking.

**Comment by richard67** [comment]
Closing as this is solved with PR #39526 .
